client/components/marketing-survey/cancel-purchase-form/index.jsx

```diff
@@ -126,6 +148,49 @@ class CancelPurchaseForm extends React.Component {
 		this.props.onInputChange( newState );
 	};
 
+	getSurveyDataType = () => {
+		if ( this.props.flowType === 'remove' ) {
+			return 'remove';
+		}
+
+		return isRefundable( this.props.purchase ) ? 'refund' : 'cancel-autorenew';
+	};
+
+	onSubmit = () => {
+		const { purchase, selectedSite } = this.props;
+
+		if ( ! isDomainRegistration( purchase ) && ! isGoogleApps( purchase ) ) {
+			this.setState( {
+				isSubmitting: true,
+			} );
+
+			const surveyData = {
+				'why-cancel': {
+					response: this.state.questionOneRadio,
+					text: this.state.questionOneText,
+				},
+				'next-adventure': {
+					response: this.state.questionTwoRadio,
+					text: this.state.questionTwoText,
+				},
+				'what-better': { text: this.state.questionThreeText },
+				type: this.getSurveyDataType(),
+			};
+
+			submitSurvey(
+				'calypso-remove-purchase',
+				selectedSite.ID,
+				enrichedSurveyData( surveyData, moment(), selectedSite, purchase )
```

Comment:
Should have `this.props.moment()` available when component has been wrapped in `localize`.

Reply:
We'd eventually like to remove `moment` from `i18-calypso` in order to avoid pulling in a big dependency that's not used most of the time. There is a new single-purpose `withLocalizedMoment` HOC and `useLocalizedMoment` hook to get `moment`.

But in this specific case, we don't use moment.js to format localized dates: just to compute a diff between two moments in `enrichedSurveyData`. For that purpose, it's best to `import moment from 'moment'` directly. No need for any wrappers or special libraries.

Reply:
Thanks for the educational discussion above, @sirreal @jsnajdr . Actually I was simply migrating what was there when I worked on this. Thus it is so great to have the valuable insights from you two 🙇🏻